File: ssssujini99/Programmers/2022_KAKAO_BLIND_RECRUITMENT/2.cpp
#include <bits/stdc++.h>

using namespace std;

bool isPrime(long long num){
    if(num<2) return false;
    for(long long i=2; i<=sqrt(num); i++){
        if(num%i==0) return false;
    }
    return true;
}


int solution(int n, int k) {
    string s = ""; int result = 0;

    while(n){
        s += (n%k + '0');
        n = n/k;
    }
    reverse(s.begin(), s.end());

    string tmp = "";

    for(int i=0; i<s.size(); i++){
        while(s[i] != '0'){
            tmp += s[i++];
            if(i == s.size()) break;
        }

        if(s[i] == '0' && tmp.size()){
            long long nn = stoll(tmp);
            if(isPrime(nn)) result++;
            tmp = "";
        }
    }

    if(tmp.size()){ // 마지막 예외처리
        long long nn = stoll(tmp);
        if(isPrime(nn)) result++;
    }

    return result;

}

File: suy2on/Programmers/2020kakao/2.py
## () 개수가 같으면 -> 균형잡힌 괄호 문자열
## () 짝도 맞다면 -> 올바른 괄호 문자열
def opposite(w):
    result = ""
    for c in w:
        if c == "(":
            result += ")"
        else:
            result += "("

    return result


def is_right(w):
    a, b = 0, 0
    for c in w:
        if c == "(":
            a += 1
        else:
            b += 1

        if a < b:
            return False

    return True


def solution(p):
    if not p:
        return p
    a, b = 0, 0
    i = 0
    while i < len(p):
        if p[i] == "(":
            a += 1
        else:
            b += 1

        i += 1
        if a == b:
            break

    u, v = p[0:i], p[i:]

    if is_right(u):
        return u + solution(v)
    else:
        return "(" + solution(v) + ")" + opposite(u[1:-1])




